### PR TITLE
Fixed NullpointerException 330

### PR DESCRIPTION
**Pull request by ciegler**
Fixed NullpointerException in https://github.com/opencast/opencast/pull/5810

### CODE REVIEW

File: modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java [hunk 1]
```diff
@@ -1641,8 +1642,8 @@ public Response getEventPublications(@HeaderParam("Accept") String acceptHeader,
 
   private List<JValue> getPublications(Event event, Boolean withSignedUrls, Boolean includeInternalPublication, ApiVersion requestedVersion) {
         return event.getPublications().stream()
-        .filter(publication -> ((includeInternalPublication && !requestedVersion.isSmallerThan(VERSION_1_11_0)) || EventUtils.internalChannelFilter.apply(publication)))
-        .map(p -> getPublication(p, withSignedUrls, requestedVersion))
+            .filter(publication -> ((includeInternalPublication != null && includeInternalPublication && !requestedVersion.isSmallerThan(VERSION_1_10_0)) || EventUtils.internalChannelFilter.apply(publication)))
```
Comment (KatrinIhler) [inline modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java]: The version is off, most likely a copy-paste error from the client repo where 1.11 doesn't exist yet. :sweat_smile:

File: modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
 import static com.entwinemedia.fn.data.json.Jsons.obj;
 import static com.entwinemedia.fn.data.json.Jsons.v;
 import static org.apache.commons.lang3.StringUtils.trimToNull;
+import static org.opencastproject.external.common.ApiVersion.VERSION_1_10_0;
```
Comment (KatrinIhler) [inline modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java]: See below.

File: modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java [hunk 3]
```diff
@@ -1641,8 +1642,8 @@ public Response getEventPublications(@HeaderParam("Accept") String acceptHeader,
 
   private List<JValue> getPublications(Event event, Boolean withSignedUrls, Boolean includeInternalPublication, ApiVersion requestedVersion) {
         return event.getPublications().stream()
-        .filter(publication -> ((includeInternalPublication && !requestedVersion.isSmallerThan(VERSION_1_11_0)) || EventUtils.internalChannelFilter.apply(publication)))
-        .map(p -> getPublication(p, withSignedUrls, requestedVersion))
+            .filter(publication -> ((includeInternalPublication != null && includeInternalPublication && !requestedVersion.isSmallerThan(VERSION_1_10_0)) || EventUtils.internalChannelFilter.apply(publication)))
+            .map(p -> getPublication(p, withSignedUrls, requestedVersion))
```
Comment (KatrinIhler) [inline modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java]: Also indentation isn't consistent. :thinking:

Comment (KatrinIhler) [reply modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java]: Still isn't consistent with the last line. *nitpick*

**Comment by KatrinIhler** [comment]
Otherwise looks good, but since this is basically my patch, somebody else should probably review this.

File: modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java [hunk 4]
```diff
@@ -1641,8 +1641,8 @@ public Response getEventPublications(@HeaderParam("Accept") String acceptHeader,
 
   private List<JValue> getPublications(Event event, Boolean withSignedUrls, Boolean includeInternalPublication, ApiVersion requestedVersion) {
         return event.getPublications().stream()
-        .filter(publication -> ((includeInternalPublication && !requestedVersion.isSmallerThan(VERSION_1_11_0)) || EventUtils.internalChannelFilter.apply(publication)))
-        .map(p -> getPublication(p, withSignedUrls, requestedVersion))
+          .filter(publication -> ((includeInternalPublication != null && includeInternalPublication && !requestedVersion.isSmallerThan(VERSION_1_11_0)) || EventUtils.internalChannelFilter.apply(publication)))
+          .map(p -> getPublication(p, withSignedUrls, requestedVersion))
         .collect(Collectors.toList());
```
Comment (mtneug) [inline modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java:1646]: As @KatrinIhler said, could you indent this line to be consistent with the two lines above?

**Review by mtneug — CHANGES_REQUESTED** [review]
This PR fixes the problem. Could you fix the code format issue? It can then go in.

File: modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java (hunk 4 above)
Comment (ciegler) [reply modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java:1646]: I am sorry, I thought I already did it, but actually I forgot to address this. The format should now be consistent.

**Review by mtneug — APPROVED** [review]

**Pull request merged**